### PR TITLE
♻️ Refine icon font detection

### PR DESCRIPTION
**Pull request by lluerich**
* adds selectors
* selector is now case-insensitive
* print warning instead of failure
* update copy to match other warnings

### CODE REVIEW

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 1]
```diff
@@ -26,6 +26,14 @@ const ICON_FONT_IDENTIFIERS = [
     className: "ociton-",
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:26]: typo? `ociton`

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 2]
```diff
@@ -26,6 +26,14 @@ const ICON_FONT_IDENTIFIERS = [
...
     fontFamilies: ["Octicons"],
   },
+  {
+    className: "icn-",
+    fontFamilies: ["icon"],
+  },
+  {
+    className: "icon-",
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:34]: This is a super generic name which I wouldn't warn about without the font name.

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 3]
```diff
@@ -35,7 +43,7 @@ export class NoIconFontIsUsed extends Rule {
   run({ $ }: Context) {
     // check for known classnames
     const iconFontCandidates = ICON_FONT_IDENTIFIERS.filter((identifier) => {
-      return $(`[class*=${identifier.className}]`).length > 0;
+      return $(`[class*=${identifier.className} i]`).length > 0;
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:46]: Out of curiosity, what does the `i` do here?

Comment (sebastianbenz) [reply packages/linter/src/rules/NoIconFontIsUsed.ts:46]: case sensitivity?

Comment (lluerich) [reply packages/linter/src/rules/NoIconFontIsUsed.ts:46]: Exactly :)

File: packages/linter/src/rules/NoIconFontIsUsed.ts (hunk 1 above)
Comment (lluerich) [reply packages/linter/src/rules/NoIconFontIsUsed.ts]: Oh sorry, fixed.

**Review by sebastianbenz — APPROVED** [review]

**Pull request merged**